pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -24,132 +24,153 @@ import (
 		issuerNamespaceName := types.NamespacedName{
 			Namespace: req.Namespace,
-			Name:      cr.Spec.IssuerRef.Name,
+			Name:      certificateRequest.Spec.IssuerRef.Name,
 		}
 		if err := r.Client.Get(ctx, issuerNamespaceName, &issuer); err != nil {
-			log.Error(err, "failed to retrieve GoogleCASIssuer resource", "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Failed to retrieve CAS Issuer resource %s: %v", issuerNamespaceName, err)
 			return ctrl.Result{}, err
 		}
 		parent = fmt.Sprintf("projects/%s/locations/%s/certificateAuthorities/%s", issuer.Spec.Project, issuer.Spec.Location, issuer.Spec.CertificateAuthorityID)
 		casClient, ok := issuerctrl.Issuers.Load(issuerNamespaceName)
 		if !ok {
 			err := errors.New("couldn't retrieve CAS Issuer client - it probably hasn't reconciled yet")
-			log.Error(err, err.Error(), "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Couldn't retrieve CAS Issuer client %s: %v", issuerNamespaceName, errors.New("couldn't retrieve CAS Issuer client"))
+			log.Error(err, err.Error(), "namespace", req.Namespace, "name", certificateRequest.Spec.IssuerRef.Name)
 			return ctrl.Result{}, err
 		}
 		googleCASClient, ok = casClient.(cas.Client)
 		if !ok {
 			err := errors.New("Retrieved a non-CAS client from the cache, this should never happen")
-			log.Error(err, err.Error(), "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Couldn't retrieve CAS Issuer client %s: %v", issuerNamespaceName, errors.New("couldn't retrieve CAS Issuer client"))
+			log.Error(err, err.Error(), "namespace", req.Namespace, "name", certificateRequest.Spec.IssuerRef.Name)
 			return ctrl.Result{}, err
 		}
 	default:
-		log.Info("Noticed unhandled kind in Certificate Request:", "kind", cr.Spec.IssuerRef.Kind)
+		log.Info("Noticed unhandled kind in Certificate Request:", "kind", certificateRequest.Spec.IssuerRef.Kind)
 		return ctrl.Result{}, nil
 	}
 
 	// Issue cert here
 	opts := &cas.CreateCertificateOptions{
 		Parent:        parent,
 		CertificateID: fmt.Sprintf("cert-manager-%d", rand.Int()),
-		CSR:           cr.Spec.Request,
-		Expiry:        cr.Spec.Duration.Duration,
+		CSR:           certificateRequest.Spec.Request,
+		Expiry:        certificateRequest.Spec.Duration.Duration,
 	}
 
 	cert, chain, err := googleCASClient.CreateCertificate(opts)
 	if err != nil {
-		log.Error(err, "Couldn't sign certificate", "namespace", req.Namespace, "name", cr.Name, "err", err)
-		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Couldn't sign certificate %s: %v", cr.Name, err)
+		log.Error(err, "Couldn't sign certificate", "namespace", req.Namespace, "name", certificateRequest.Name, "err", err)
```

Comment:
Is there any way to distinguish between temporary and permanent errors here? Like transient network errors that should be retried by returning the error vs Google CAS validataion errors which are probably permanent until the user alters the certiticate / cetificaterequest spec.